src/App.tsx

```diff
@@ -1,26 +1,416 @@
-
-const USER_ID = 0;
+import {
+  USER_ID,
+  getTodos,
+  createTodos,
+  deleteTodos,
+  updateTodos,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterStatus } from './types/FilterStatus';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoList } from './components/TodoList';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState('');
+  const [filterStatus, setFilterStatus] = useState(FilterStatus.All);
+
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [isAddingTodo, setIsAddingTodo] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [deletingTodoIds, setDeletingTodoIds] = useState<number[]>([]);
+
+  const [updatingTodoIds, setUpdatingTodoIds] = useState<number[]>([]);
+
+  const newTodoFieldRef = React.useRef<HTMLInputElement>(null);
+
+  const [editingTodoId, setEditingTodoId] = useState<number | undefined>(
+    undefined,
+  );
+
+  const [updatedTodoId, setUpdatedTodoId] = useState<number | undefined>(
+    undefined,
+  );
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      try {
+        setIsLoading(true);
+        setError('');
+
+        const loadedTodos = await getTodos();
+
+        setTodos(loadedTodos);
+      } catch (e) {
+        setError('Unable to load todos');
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    loadTodos();
+  }, []);
+
+  useEffect(() => {
+    if (error) {
+      const timerId = setTimeout(() => {
+        setError('');
+      }, 3000);
+
+      return () => {
+        clearTimeout(timerId);
+      };
+    }
+
+    return undefined;
+  }, [error]);
+
+  useEffect(() => {
+    if (!isAddingTodo && newTodoFieldRef.current) {
+      newTodoFieldRef.current.focus();
+    }
+  }, [isAddingTodo]);
+
+  useEffect(() => {
+    if (updatedTodoId !== undefined) {
+      const timerId = setTimeout(() => {
+        setUpdatedTodoId(undefined);
+      }, 100);
+
+      return () => {
+        clearTimeout(timerId);
+      };
+    }
+
+    return undefined;
+  }, [updatedTodoId]);
+
+  const hideErrorNotification = () => {
+    setError('');
+  };
+
+  const handleTodoSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const trimmedTitle = newTodoTitle.trim();
+
+    if (!trimmedTitle) {
+      setError('Title should not be empty');
+
+      return;
+    }
+
+    try {
+      setIsAddingTodo(true);
+
+      setTempTodo({
+        id: 0,
+        userId: USER_ID,
+        title: trimmedTitle,
+        completed: false,
+      });
+
+      const newTodo = await createTodos(trimmedTitle);
+
+      setTodos(prevTodos => [...prevTodos, newTodo]);
+      setNewTodoTitle('');
+    } catch (e) {
+      setError('Unable to add a todo');
```

Comment:
don't rely on unknown string, create enum with error messages and use it instead of magic string